postgres/datadog_checks/postgres/util.py

```diff
@@ -172,21 +172,28 @@ def get_schema_field(descriptors):
 REPLICATION_STATS_METRICS = {
     'descriptors': [('application_name', 'wal_app_name'), ('state', 'wal_state'), ('sync_state', 'wal_sync_state')],
     'metrics': {
-        'GREATEST (0, EXTRACT(epoch from write_lag)) as write_lag': (
+        'max(GREATEST (0, EXTRACT(epoch from write_lag))) as write_lag': (
             'postgresql.replication.wal_write_lag',
             AgentCheck.gauge,
         ),
-        'GREATEST (0, EXTRACT(epoch from flush_lag)) AS flush_lag': (
+        'max(GREATEST (0, EXTRACT(epoch from flush_lag))) AS flush_lag': (
             'postgresql.replication.wal_flush_lag',
             AgentCheck.gauge,
         ),
-        'GREATEST (0, EXTRACT(epoch from replay_lag)) AS replay_lag': (
+        'max(GREATEST (0, EXTRACT(epoch from replay_lag))) AS replay_lag': (
             'postgresql.replication.wal_replay_lag',
             AgentCheck.gauge,
         ),
+        'max(GREATEST (0, age(backend_xmin))) as backend_xmin_age': (
+            'postgresql.replication.backend_xmin_age',
+            AgentCheck.gauge,
+        ),
     },
     'relation': False,
-    'query': 'SELECT application_name, state, sync_state, {metrics_columns} FROM pg_stat_replication',
+    'query': """
+SELECT application_name, state, sync_state, {metrics_columns}
+FROM pg_stat_replication
+GROUP BY application_name, state, sync_state""",
```

Comment:
Do you know what was happening to duplicate `application_name, state, sync_state` values before adding the `group by`?  It seems like we're losing some information here, since multiple WAL senders can have the same `application_name, state, sync_state`.

I think a better solution might be to add `pid` as a column instead of doing a group by.  That way we won't group multiple wal senders into a single row, and we'll be able to get the max values via metrics query.

Reply:
Yeah, I was hesitating between the aggregation or adding another tag.
As a way to differentiate values, `client_addr` might be better. This will allow to pinpoint the lagging replica while being more stable than pid.

Reply:
Besides that, pid could have the same value on different servers.